third_party/move/move-binary-format/src/errors.rs

```diff
@@ -659,3 +659,13 @@ impl std::error::Error for PartialVMError {
         None
     }
 }
+
+pub trait PartialVMResultExt<T> {
```

Comment:
You can use VMResult instead? and convert `PartialVMResult` into it lie:
```
let e = PartialVMError::new(...);
let vm_error = e.finish(your_location)
```
Why have a trait? Also `Ext` reads ugly :)

Reply:
I don't quite get your point. This is just a shorter version for `.map_err(|err| err.finish(location)`.

The `PartialVMResult` is generated somewhere down the stack and propagates up to some point where we need to convert it to the `VMResult`, and today we do it with `.map_err()` which is verbose and annoys me -- I mean, I can't "use VMResult instead".

an `Ext` trait is something I learnt from the Rust async/futures stack https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html which I thought was neat. 

obviously I don't feel strongly about this. lmk you do -- I can just type a few characters more and use `.map_err()`

Reply:
The whole `PartialVMResult` and `VMResult` distinction is pretty much broken :( As `VMResult` carries the location and we process outside of the VM it will always be undefined....

Neat, was not aware of this [https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html ](https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html.)

My only concern here tbh is that we have so many different types/traits that do the same thing, we should probably avoid adding more and instead get rid of the ugliness of existing error system, rather than trying to find shortcuts. The more annoying it gets, the more motivation there is to fix it

Reply:
tell me about it being "pretty much broken". :trollface: 

will remove the ext trait.